### PR TITLE
OSv2 Mule 4 Tutorial update

### PR DESCRIPTION
**Pull request by george-hoffer**

### CODE REVIEW

File: modules/ROOT/pages/osv2-tutorial.adoc [hunk 1]
```diff
@@ -5,20 +5,339 @@ endif::[]
...
 
-This tutorial works with Studio 6 using Mule 3.
-
 == Prerequisites
 
-This document assumes you are familiar with Anypoint Studio, using curl from a command line, and the use of Anypoint Platform and Runtime Manager. For more resources, view the See Also section of this document.
+This document assumes you are familiar with Anypoint Studio, using curl from a command line, 
+and the use of Anypoint Platform and Runtime Manager. Ensure that curl is installed on 
+your computer. If you are using a Mac or Linux, the curl command is built in. For Windows, 
+download and install curl from the net.
```
Comment (jjayakumaran) [inline modules/ROOT/pages/osv2-tutorial.adoc]: will be good to link a download link to curl, like we do in mq tutorial

File: modules/ROOT/pages/osv2-tutorial.adoc [hunk 2]
```diff
@@ -5,20 +5,339 @@ endif::[]
...
+
+The output from this command lists the value for the key:
+
+[source,console]
+----
+"OSv2 Test"
+----
+
+From this exercise, the Object Store user interface helps you
+see that the application you created in Design Center can be accessed using a curl command.
```
Comment (jjayakumaran) [inline modules/ROOT/pages/osv2-tutorial.adoc]: not the application itself, but the OSv2 linked to your application can be accessed using a curl command.

File: modules/ROOT/pages/osv2-tutorial.adoc (hunk 1 above)
Comment (jjayakumaran) [reply modules/ROOT/pages/osv2-tutorial.adoc]: also we can use postman or browser extensions, besides curl

**Review by jjayakumaran — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/osv2-tutorial.adoc (hunk 2 above)
Comment (george-hoffer) [reply modules/ROOT/pages/osv2-tutorial.adoc]: Changed to:
For future use of Object Store in Design Center, you don't need to view the Runtime Manager interface each time you update the Object Store. After you create an application linked to the OSv2 connector, you can store a key and value in the Object Store with one curl command and view the results with another. Just change the key and/or the value as needed.

File: modules/ROOT/pages/osv2-tutorial.adoc [hunk 3]
```diff
@@ -5,20 +5,339 @@ endif::[]
...
 
 . In Anypoint Studio, click the Exchange icon in the Studio task bar.
 . Click Login in Anypoint Exchange.
 . Search for the "objectstore" connector and click Install.
 . Follow the prompts to install the connector.
 
-== Create a Studio Project
+== Create a Design Center Application
+
+IMPORTANT: Disable password programs such as LastPass or Okta in your browser before using Design Center. Otherwise, these programs inserts passwords in the HTTP Listener’s TLS tab. The passwords in the TLS tab cause your application's deployment to fail.
```
Comment (steveCluster) [inline modules/ROOT/pages/osv2-tutorial.adoc:27]: Can you check with the flow designer team if this password bug is still present in design center?  If not we can remove this warning.

File: modules/ROOT/pages/osv2-tutorial.adoc [hunk 4]
```diff
@@ -5,20 +5,339 @@ endif::[]
...
+. In *Select a component*, search for "Obj", and click the *ObjectStore Connector*.
+. Click the *Retrieve* operation and in the Key field, specify the `#[attributes.queryParams.key]` value.
+
+This completes the Retrieve flow. Click Run again to restart your application.
+
+In the command prompt, submit this curl command to view the key's value:
+
+[source,console]
+----
+curl http://osv2-qoav.us-w1.cloudhub.io/retrieve?key=MyKey;echo
```
Comment (steveCluster) [inline modules/ROOT/pages/osv2-tutorial.adoc]: Doesn't this link need to change based on the customer's application name?  Shouldn't we use the same approach as was used above where we make the user copy in the link from design center as LINK_FROM_DC in the url?

File: modules/ROOT/pages/osv2-tutorial.adoc [hunk 5]
```diff
@@ -5,20 +5,339 @@ endif::[]
...
+[source,console]
+----
+curl -X POST -H "Content-Type: application/json" -d '{ "key": "MyKey", "value": "OSv2 Test" }' "http://osv2-example.us-w1.cloudhub.io/store"
+----
+
+Retrieve:
+
+[source,console]
+----
+curl http://osv2-qoav.us-w1.cloudhub.io/retrieve?key=MyKey
```
Comment (steveCluster) [inline modules/ROOT/pages/osv2-tutorial.adoc]: Same thing here:
Doesn't this link need to change based on the customer's application name?  Shouldn't we use the same approach as was used above where we make the user copy in the link from design center as LINK_FROM_DC in the url?

File: modules/ROOT/pages/osv2-tutorial.adoc [hunk 6]
```diff
@@ -5,20 +5,339 @@ endif::[]
...
++
+You can see that `MyKey` made it into the Object Store user interface, which confirms how to
+view that keys successfully can be sent through your Design Center application and stored in 
+the Object Store.
++
+. To view the value for the key, use this curl command:
++
+[source,console]
+----
+curl http://osv2-qoav.us-w1.cloudhub.io/retrieve?key=MyKey
```
Comment (steveCluster) [inline modules/ROOT/pages/osv2-tutorial.adoc]: And again here:
Doesn't this link need to change based on the customer's application name?  Shouldn't we use the same approach as was used above where we make the user copy in the link from design center as LINK_FROM_DC in the url?

File: modules/ROOT/pages/osv2-tutorial.adoc [hunk 7]
```diff
@@ -7,10 +7,10 @@ This document helps you use Object Store v2 (OSv2) to store and view data.
 
 == Prerequisites
 
-This document assumes you are familiar with Anypoint Studio, using curl from a command line, 
+This document assumes you are familiar with Anypoint Studio, using https://curl.haxx.se/[curl] from a command line, 
 and the use of Anypoint Platform and Runtime Manager. Ensure that curl is installed on 
 your computer. If you are using a Mac or Linux, the curl command is built in. For Windows, 
-download and install curl from the net.
+download and install curl from the net. Alternatively, you can use Postman or browser plugins instead of curl.
```
Comment (jjayakumaran) [inline modules/ROOT/pages/osv2-tutorial.adoc]: can we just say download and install curl? and remove **from the net**

File: modules/ROOT/pages/osv2-tutorial.adoc (hunk 3 above)
Comment (george-hoffer) [reply modules/ROOT/pages/osv2-tutorial.adoc:27]: This bug is still in force. The flow designer team may fix it in the future but for documenting now, this must be said or users with LP/Okta deployments will fail every time. You can't just delete the passwords when these services operate -- the apps automatically fill in the TLS passwords as soon as focus changes.

File: modules/ROOT/pages/osv2-tutorial.adoc (hunk 4 above)
Comment (george-hoffer) [reply modules/ROOT/pages/osv2-tutorial.adoc]: Fixed throughout -- the curl commands now have either LINK_FROM_DESIGN_CENTER or LINK_FROM_RUNTIME_MANAGER depending on the section.

File: modules/ROOT/pages/osv2-tutorial.adoc (hunk 6 above)
Comment (george-hoffer) [reply modules/ROOT/pages/osv2-tutorial.adoc]: Fixed as requested.

File: modules/ROOT/pages/osv2-tutorial.adoc (hunk 7 above)
Comment (george-hoffer) [reply modules/ROOT/pages/osv2-tutorial.adoc]: Fixed as requested.

File: modules/ROOT/pages/osv2-tutorial.adoc (hunk 5 above)
Comment (george-hoffer) [reply modules/ROOT/pages/osv2-tutorial.adoc]: Fixed. The curl commands now have either LINK_FROM_DESIGN_CENTER or LINK_FROM_RUNTIME_MANAGER depending on the section.

**Review by jjayakumaran — APPROVED** [review]

**Pull request merged**